feat(genapi): integrating generative apis with popular tools (#4649)
scaleway/docs-content: pages/generative-apis/reference-content/integrating-generative-apis-with-popular-tools.mdx

| [OpenAI client](#openai-client-libraries) | Popular AI library for natural language processing | Text generation, language translation, text summarization | Low |
| [LangChain](#langchain-rag-and-llm-applications) | Library for building AI applications leveraging RAG | Inference, embeddings, document indexing and retrieval | Medium |
| [LlamaIndex](#llamaindex-advanced-rag-applications) | Library for building advanced AI RAG applications | Knowledge graph building, document retrieval, data indexing | Medium |

> | [Continue Dev](#continue-dev-ai-coding-assistance) | IDE Extension for AI-powered coding assistance | Code completion, code review | Low |

Reviewer: ```suggestion
| [Continue Dev](#continue-dev-ai-coding-assistance) | IDE extension for AI-powered coding assistance | Code completion, code review | Low |
```

<Macro id="requirements" />

- A Scaleway account logged into the [console](https://console.scaleway.com)
- [Owner](/iam/concepts/#owner) status or [IAM permissions](/iam/concepts/#permission) allowing you to perform actions in the intended Organization
- A valid [API key](/iam/how-to/create-api-keys/) for API authentication
- Installed [Zed](https://zed.dev/) on your local machine

### Configure custom endpoints and models

1. Edit Zed settings located in `settings.json`, and add the following content to it:
```json
{
  "language_models": {
    "openai": {
      "api_url": "https://api.scaleway.ai/v1",
      "available_models": [
        {
          "name": "qwen2.5-coder-32b-instruct",
          "display_name": "Qwen 2.5 Coder 32B",
          "max_tokens": 128000
        }
      ],
      "version": "1"
    }
  },
  "assistant": {
    "default_model": {
      "provider": "openai",
      "model": "qwen2.5-coder-32b-instruct"
    },
    "version": "2"
  }
}
```

This configuration will add a `qwen2.5-coder-32b-instruct` Scaleway hosted model available with the Zed `openai` provider, and use it as default model.

> 2. Open AI Assistant configuration by either using the command palette and typing `assistant: show configuration` or clicking on the bottom right **Assistant Panel** button and then **Assistant menu** in top right and finally **Configure**.

Reviewer: There seems to be a verb missing here

> 3. Scroll down to the OpenAI Configuration, and paste your Scaleway Secret Key as API Key credentials. Note that this key will be deleted if you restart Zed. To store it permanently, alternatively, you can also setup your Scaleway Secret Key as `OPENAI_API_KEY` environment variable and restart Zed.

Reviewer: ```suggestion
3. Scroll down to the OpenAI Configuration, and paste your Scaleway secret key as API Key credentials. Note that this key will be deleted if you restart Zed. To store it permanently, set up your Scaleway secret key as `OPENAI_API_KEY` environment variable and restart Zed.
```

> Your setup is complete. If you open a new chat and select `Qwen 2.5 Coder 32B` model, you can send text and retrieve model answer. Additionnaly, you can also use [Inline Assist](https://zed.dev/docs/assistant/inline-assistant#:~:text=You%20can%20use%20ctrl%2Denter,even%20within%20the%20terminal%20panel.) feature when editing your code.

Reviewer: ```suggestion
Your setup is complete. If you open a new chat and select `Qwen 2.5 Coder 32B` model, you can send text and retrieve model answers. Additionally, you can also use [Inline Assist](https://zed.dev/docs/assistant/inline-assistant#:~:text=You%20can%20use%20ctrl%2Denter,even%20within%20the%20terminal%20panel.) feature when editing your code.
```

> 4. Your setup is complete. If you open a new chat and select `Qwen 2.5 Coder 32B` model, you can send text and retrieve model answer. Additionnaly, you can also use [Inline Assist](https://zed.dev/docs/assistant/inline-assistant#:~:text=You%20can%20use%20ctrl%2Denter,even%20within%20the%20terminal%20panel.) feature when editing your code.

Reviewer: ```suggestion
Your setup is complete. If you open a new chat and select `Qwen 2.5 Coder 32B` model, you can send text and retrieve model answer. Additionnaly, you can also use [Inline Assist](https://zed.dev/docs/assistant/inline-assistant#:~:text=You%20can%20use%20ctrl%2Denter,even%20within%20the%20terminal%20panel.) feature when editing your code.
```
